src/directory/utils.ts

```diff
@@ -163,3 +198,156 @@ export function isRawFileMetadata(value: unknown): value is RawFileMetadata {
+  if (files.length === 0) {
+    return []
+  }
+
+  const testFilePath = files[0]?.webkitRelativePath
+  assertString(testFilePath, '"webkitRelativePath" property should be a string')
+  const parts = testFilePath.split('/')
+
+  // `webkitRelativePath` always contains base file path
+  if (parts.length < 2) {
+    throw new Error(`"webkitRelativePath" does not contain base path part: "${testFilePath}"`)
+  }
+
+  return Array.from(files).map(file => {
+    const relativePath = file.webkitRelativePath.substring(parts[0].length + 1)
+
+    return {
+      fileSystemType: FileSystemType.browser,
+      fullPath: '',
+      relativePath,
+      relativePathWithBase: file.webkitRelativePath,
+      browserFile: file,
+    }
+  })
+}
+
+/**
+ * Gets files list with base path like in a browser's `File` object
+ */
+export async function getNodeFileInfoList(path: string, recursive: boolean): Promise<FileInfo[]> {
+  const paths = await getNodePaths(path, recursive)
+  const pathLength = path.length + 1
+  const basePath = nodePath.basename(path)
+
+  return paths.map(fullPath => {
+    const relativePath = fullPath.substring(pathLength)
+    const relativePathWithBase = nodePath.join(basePath, relativePath)
+
+    return {
+      fileSystemType: FileSystemType.node,
+      fullPath,
+      relativePath,
+      relativePathWithBase,
+    }
+  })
+}
+
+/**
+ * Assert that `File` instance from browser contains `webkitRelativePath`
+ */
+export function assertBrowserFilesWithPath(value: unknown): asserts value is FileList {
+  if (isNode()) {
+    throw new Error('File info asserting is available only in browser')
+  }
+
+  if (!(value instanceof FileList)) {
+    throw new Error('Browser files is not `FileList`')
+  }
+
+  const data = Array.from(value)
+  for (const item of data) {
+    if (!(item instanceof File)) {
+      throw new Error(`Item of browser files is not a File instance`)
+    }
+
+    if (!('webkitRelativePath' in item)) {
+      throw new Error(`${item.name} does not contain "webkitRelativePath"`)
+    }
+  }
+}
+
+/**
+ * Filters FileInfo items where filename starts with dot
+ */
+export function filterDotFiles(files: FileInfo[]): FileInfo[] {
+  return files.filter(item => {
+    const basename = getBaseName(item.relativePath)
+
+    return !basename || !basename.startsWith('.')
+  })
+}
+
+/**
+ * Filters extra files found recursively which browser adds by default
+ */
+export function filterBrowserRecursiveFiles(files: FileInfo[]): FileInfo[] {
+  return files.filter(item => !item.relativePath.includes('/'))
+}
+
+/**
+ * Gets files content in Node.js environment
+ */
+export function getNodeFileContent(fullPath: string): Uint8Array {
+  if (!fs.existsSync(fullPath)) {
+    throw new Error(`File does not exist: "${fullPath}"`)
+  }
+
+  return fs.readFileSync(fullPath)
+}
+
+/**
+ * Gets target absolute upload path
+ */
+export function getUploadPath(fileInfo: FileInfo, isIncludeDirectoryName: boolean): string {
```

Comment:
because I think that the source of standard behavior is better to be the options variable. All other functions should rather depend on it than have each function have its own definition of a default value.